Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -0,0 +1,198 @@
+      "inputs": {
+        "Service": "ec2",
+        "Api": "DescribeSnapshots",
+        "SnapshotIds": "{{CreateSnapshot.Output}}"
+      },
+      "outputs": [
+        {
+          "Name": "SNAPSHOTID",
+          "Selector": "$.Snapshots[0].SnapshotId",
+          "Type": "String"
+        }
+      ],
+      "name": "extractSnapshotID",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 30,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "CopySnapshot",
+        "SourceSnapshotId": "{{extractSnapshotID.SNAPSHOTID}}",
+        "SourceRegion": "{{region}}",
+        "Encrypted": true,
+        "KmsKeyId": "{{KmsKeyId}}",
+        "DestinationRegion": "{{region}}"
+      },
+      "outputs": [
+        {
+          "Name": "EncryptedSnapshotID",
+          "Selector": "$.SnapshotId",
+          "Type": "String"
+        }
+      ],
+      "name": "CopySnapshot",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 3600,
+      "onFailure": "Abort"
+    },
+    {
+      "inputs": {
+        "Duration": "PT2M"
+      },
+      "name": "sleep1",
+      "action": "aws:sleep"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "CreateVolume",
+        "AvailabilityZone": "{{extractAvailabilityZone.PlacementAvailabilityZone}}",
+        "Encrypted": true,
+        "KmsKeyId": "{{KmsKeyId}}",
+        "SnapshotId": "{{CopySnapshot.EncryptedSnapshotID}}",
+        "VolumeType": "gp2"
+      },
+      "outputs": [
+        {
+          "Name": "NewRootVolumeID",
+          "Selector": "$.VolumeId",
+          "Type": "String"
+        }
+      ],
+      "name": "CreateVolume",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 30,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "DocumentName": "AWS-StopEC2Instance",
+        "RuntimeParameters": {
+          "InstanceId": "{{instanceId}}"
+        }
+      },
+      "name": "StopInstance",
+      "action": "aws:executeAutomation",
+      "timeoutSeconds": 300,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "DocumentName": "AWS-DetachEBSVolume",
+        "RuntimeParameters": {
+          "VolumeId": "{{extractEBSvolumeID.EBSVolumeID}}"
+        }
+      },
+      "name": "DetachEBSVolume",
+      "action": "aws:executeAutomation",
+      "timeoutSeconds": 300,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "DocumentName": "AWS-AttachEBSVolume",
+        "RuntimeParameters": {
+          "Device": "{{devicename}}",
+          "InstanceId": "{{instanceId}}",
+          "VolumeId": "{{CreateVolume.NewRootVolumeID}}"
+        }
+      },
+      "name": "AttachNewEBSVolume",
+      "action": "aws:executeAutomation",
+      "timeoutSeconds": 180,
+      "onFailure": "Abort"
+    }
```

Comment:
At the end, you should restore the instance initial state (which you can describe along with the other properties you need). Since you stopped the instance to detach its root volume, you may want to start the instance again if that was the initial state, or keep it stopped if the instance was stopped to begin with.

Reply:
Great suggestion.  Done.